flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcSystem.java

```diff
@@ -25,7 +25,7 @@
  * This interface serves as a factory interface for RPC services, with some additional utilities
  * that are reliant on implementation details of the RPC service.
  */
-public interface RpcSystem extends RpcSystemUtils {
+public interface RpcSystem extends RpcSystemUtils, AutoCloseable {
```

Comment:
I'd actually always go for `Closeable` unless you need to throw checked exceptions beyond `IOException`.

The practical implications are actually quite small except that you can use Guava's `Closer` only with `Closeables` but there may also other code that still requires the more specific interface.